### PR TITLE
fix: Parse Server option `fileExtensions` default value rejects file extensions that are less than 3 or more than 4 characters long

### LINKED ISSUE #8698: Default fileExtension regex does not match `plain` files.
### New Issue Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

The current default `fileExtension` regex (`"^[^hH][^tT][^mM][^lL]?$"`) does not match `plain`.

### Steps to reproduce

1. Do not set fileExtension
2. Try to create file

### Actual Outcome

Error

### Expected Outcome

Should succeed

### Environment

Server
- Parse Server version: `FILL_THIS_OUT`
- Operating system: `FILL_THIS_OUT`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `FILL_THIS_OUT`

Database
- System (MongoDB or Postgres): `FILL_THIS_OUT`
- Database version: `FILL_THIS_OUT`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `FILL_THIS_OUT`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `FILL_THIS_OUT`
- SDK version: `FILL_THIS_OUT`

### PR DESCRIPTION
**Pull request by dblythy**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Closes: #8698

## Approach

Changes the default fileExtension regex

## Tasks

- [x] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

File: src/Options/Definitions.js [hunk 1]
```diff
@@ -1001,9 +1001,9 @@ module.exports.FileUploadOptions = {
   fileExtensions: {
     env: 'PARSE_SERVER_FILE_UPLOAD_FILE_EXTENSIONS',
     help:
-      "Sets the allowed file extensions for uploading files. The extension is defined as an array of file extensions, or a regex pattern.<br><br>It is recommended to restrict the file upload extensions as much as possible. HTML files are especially problematic as they may be used by an attacker who uploads a HTML form to look legitimate under your app's domain name, or to compromise the session token of another user via accessing the browser's local storage.<br><br>Defaults to `^[^hH][^tT][^mM][^lL]?$` which allows any file extension except HTML files.",
+      "Sets the allowed file extensions for uploading files. The extension is defined as an array of file extensions, or a regex pattern.<br><br>It is recommended to restrict the file upload extensions as much as possible. HTML files are especially problematic as they may be used by an attacker who uploads a HTML form to look legitimate under your app's domain name, or to compromise the session token of another user via accessing the browser's local storage.<br><br>Defaults to `^(?!html?$)` which allows any file extension except HTML files.",
```
Comment (mtrezza) [inline src/Options/Definitions.js]:
```suggestion
      "Sets the allowed file extensions for uploading files. The extension is defined as an array of file extensions, or a regex pattern.<br><br>It is recommended to restrict the file upload extensions as much as possible. HTML files are especially problematic as they may be used by an attacker who uploads a HTML form to look legitimate under your app's domain name, or to compromise the session token of another user via accessing the browser's local storage.<br><br>Defaults to `^(?!(h|H)(t|T)(m|M)(l|L)?$)` which allows any file extension except HTM(L) files.",
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: src/Options/Definitions.js (hunk 1 above)
Comment (mtrezza) [reply src/Options/Definitions.js]: fixed

**Comment by mtrezza** [comment]
This PR should fix errors in https://github.com/parse-community/Parse-SDK-JS/pull/2089

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.26](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.26)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.0.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0](https://github.com/parse-community/parse-server/releases/tag/7.0.0)

**Pull request merged**